assets/js/admin.js

```diff
@@ -161,15 +161,286 @@
+
+				$target.empty();
+
+				$blocks.each( function() {
+
+					dayRow.cloneTimeBlock( $( this ), $target, true, true );
+
+				} );
+
+			} );
+
+		},
+
+		cloneTimeBlock: function ( $block, $target, deep, flashDay ) {
+
+			var $day         = $block.closest( '.day-row' ),
+			    day          = parseInt( $day.attr( 'data-day' ), 10 ),
+			    target_day   = parseInt( $target.closest( '.day-row' ).attr( 'data-day' ), 10 ),
+			    block        = parseInt( $block.attr( 'data-time-block' ), 10 ),
+			    target_block = $target.is( ':empty' ) ? 0 : parseInt( $target.find( '.time-block' ).last().attr( 'data-time-block' ), 10 ) + 1,
+			    deep         = deep || false, // jshint ignore:line
+			    flashDay     = flashDay || false, // jshint ignore:line
+			    $clone       = $block.clone( deep ),
+			    html         = $clone.html(),
+			    name_search  = '\\[schedule\\]\\[' + day + '\\]\\[blocks\\]\\[' + block + '\\]',
+			    name_replace = '[schedule][' + target_day + '][blocks][' + target_block + ']',
+			    id_search    = 'schedule-' + day + '-blocks-' + block,
+			    id_replace   = 'schedule-' + target_day + '-blocks-' + target_block;
+
+			html = html.replace( new RegExp( name_search, 'g' ), name_replace ),
+			html = html.replace( new RegExp( id_search, 'g' ), id_replace ),
+
+			$clone = $clone.html( html );
+
+			if ( deep ) {
+
+				$clone.find( 'select' ).each( function() {
+
+					var classes = $( this )
+						.attr( 'class' )
+						.split( ' ' )
+						.map( function ( v ) {
+							return v.trim();
+						} )
+						.join( '.' );
+
+					$( this ).val( $block.find( 'select.' + classes ).val() );
+
+				} );
+
+			}
+
+			$target.append( $clone );
+
+			$clone.attr( 'data-time-block', target_block );
+
+			if ( flashDay ) {
+
+				$clone.closest( '.day-row' ).fadeTo( 50, 0.1, function() {
+
+					$( this ).fadeTo( 500, 1.0 );
+
+				} );
+
+			}
+
+			return $clone;
+
+		},
+
+		toggleClosed: function () {
+
+			var $day   = $( this ).closest( '.day-row' ),
+			    closed = $day.hasClass( 'status-closed' );
+
+			$day.toggleClass( 'status-closed' ).toggleClass( 'status-open' );
+
+			$day.find( 'a.button' ).toggleClass( 'disabled', ! closed );
+
+			$day.find( 'select' ).toggleClass( 'disabled', ! closed );
+
+		},
+
+		changeTime: function( e ) {
+
+			var $target        = $( e.currentTarget ),
+			    time           = $target.val();
+
+			dayRow.updateTimeSelect( $target, time );
+
+		},
+
+		updateTimeSelect: function( $target, time ) {
+
+			if ( ! $target ) {
+
+				$( document ).find( '.wpcw-widget-hours .time-block-open' ).each( function() {
+
+					dayRow.updateTimeSelect( $( this ), $( this ).val() );
+
+				} );
+
+				return;
+
+			}
+
+			var $parent = $target.closest( '.time-block' ),
+			    $close_select = $parent.find( '.time-block-close' );
+
+			$close_select.children().show();
+			$close_select.find( 'option[value="' + time + '"]' ).prevAll().andSelf().hide();
```

Comment:
@EvanHerman `.andSelf()` is deprecated and replaced by `.addBack()`

cf. https://api.jquery.com/andself/